internal/controller/helmrepository_controller.go

```diff
@@ -593,8 +586,18 @@ func (r *HelmRepositoryReconciler) reconcileArtifact(ctx context.Context, sp *pa
 	}
 	defer unlock()
 
-	// Save artifact to storage.
-	if err = r.Storage.CopyFromPath(artifact, chartRepo.Path); err != nil {
+	// Save artifact to storage in JSON format.
+	b := &bytes.Buffer{}
+	if err := json.NewEncoder(b).Encode(chartRepo.Index); err != nil {
```

Comment:
Might be better to move this to a method on `chartRepo`, in addition it would be nice to provide a bit prettier JSON output using: 

```suggestion
	b, err := json.MarshalIndent(chartRepo.Index, "", "  ")
	if err != nil {
```